ynput/ayon-core
# Publisher: Separate backend into models

## Changelog Description
Separate publisher tool logic into smaller models.

## Additional info
Content of `control.py` was huge mess with ton of public methods and properties, it was hard to find out what is used where and how. Hopefully this helps a little bit to manage the logic. Some of the changes may cause issues with custom modifications of the tool (e.g. traypublisher). Some methods/properties were kept to keep backwards compatibility.

This is first PR that is "refactoring" the logic. There will be more. This is already huge.

## Testing notes:
Publisher tool should work as usual in host integrations.
1. Validate any feature of publisher.
- create
- publish
- pause publishing
- see validation errors
- see errors
- change attributes
- change context
- save should keep values
- invalid instance context should not cause crash (which did in develop)
- reset should do what did reset do
- buttons are enabled/disabled when they should be
- publish report can be loaded in publish report viewer
